tests/unit/test-extension-runners/test.firefox-android.js

```diff
@@ -704,37 +704,99 @@ describe('util/extension-runners/firefox-android', () => {
          sinon.assert.calledOnce(anotherCallback);
        });
 
-    it('checks the granted android permissions on Android >= 21',
+    it('raised an error when unable to find an android version number',
        async () => {
-         const {
-           params, fakeADBUtils,
-         } = prepareSelectedDeviceAndAPKParams();
+         async function expectInvalidVersionError(version: any) {
+           const {
+             params, fakeADBUtils,
+           } = prepareSelectedDeviceAndAPKParams();
 
-         fakeADBUtils.getAndroidVersionNumber = sinon.spy(() => {
-           Promise.resolve(21);
-         });
+           fakeADBUtils.getAndroidVersionNumber = sinon.spy(() => {
+             return version;
+           });
 
-         const runnerInstance = new FirefoxAndroidExtensionRunner(params);
+           const runnerInstance = new FirefoxAndroidExtensionRunner(params);
+           const promise = runnerInstance.run();
```

Comment:
Shouldn't this be `const promise = await runnerInstance.run()` so that `assert.isRejected()` gets the finished promise? I can't quite tell from the chai-as-promised docs how it works. Alternatively, maybe `expectInvalidVersionError()` needs to return `promise` so that the test always waits for completion?

Reply:
ah, great catch!

You are right, I forgot to re-add the await on the `assert.isRejected(..)` calls when I refactored the test to make it easier to test all the values that I wanted to test, the right way to use it is like we are using it in the existent test.adb.js tests:

https://github.com/mozilla/web-ext/blob/1b22d60aefa31e2869e15d18ec52b39afa0de405/tests/unit/test-util/test.adb.js#L106-L107

(unfortunately the current version of the chai-as-promised docs do not have any example that uses the async functions, on the contrary Jest provides a similar `rejects` assertion and it also provide an example of its usage in an async function: https://facebook.github.io/jest/docs/en/expect.html#rejects)

We can't use `await runnerInstance.run(...)` because we expect a rejection and the assertion would never be reached.